Modules/functions.c

```diff
@@ -30,6 +30,75 @@ l_ldap_initialize(PyObject *unused, PyObject *args)
     return (PyObject *)newLDAPObject(ld);
 }
 
+#ifdef HAVE_LDAP_INIT_FD
+
+/* initialize_fd(fileno, url)
+ *
+ * ldap_init_fd() is not a private API but it's not in a public header either
+ * SSSD has been using the function for a while, so it's probably OK.
```

Comment:
Are there any plans to make it public?

Reply:
I don't know why it hasn't landed in upstream headers. The API has been around for a while and it's stable. SSSD uses it heavily.

Reply:
Is there an RFE open?
I'm worried by the extern declaration: OpenLDAP is free to add a `ldap_init_fd` function to the public headers, and even make the declaration a bit different. (Would e.g. `unsigned int proto` be enough to cause python-ldap to fail to build?).

Reply:
It's been document with the exact same function signature for 13 years, https://github.com/openldap/openldap/commit/282a3c537a9ab2f805a2f428606a3da3fa1705d5 . It's unlikely that the function will disappear or change signature.

Reply:
OK.
Could you make sure OpenLDAP devs know, though?

Reply:
> 
> 
> OK.
> Could you make sure OpenLDAP devs know, though?

The statement was false.  It is provided in the public openldap.h header file.

Reply:
... which is not packaged by distributions like Debian/Ubuntu or Fedora/RHEL.

Reply:
> 
> 
> ... which is not packaged by most distributions like Debian/Ubuntu or Fedora/RHEL.

Then you need to talk to the distributions, not the OpenLDAP developers.